### PR TITLE
[TACHYON-457] Move localfs from common to a submodule of underfs

### PR DESCRIPTION
**Pull request by apc999**
https://tachyon.atlassian.net/browse/TACHYON-457

Summary:
This revision moves tachyon.underfs.local from tachyon-common to tachyon-underfs
with other under file system implementations. The advantage is the code structure is cleaner.

Test Plan: Unit-tests and integration tests passed

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -78,6 +78,12 @@
       <version>${project.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.tachyonproject</groupId>
+      <artifactId>tachyon-underfs-local</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (calvinjia) [inline integration-tests/pom.xml]: Do we still need the test-jar from `tachyon-servers` after this change?

File: underfs/local/src/test/java/tachyon/underfs/local/LocalUnderFileSystemFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import tachyon.conf.TachyonConf;
+import tachyon.underfs.UnderFileSystemFactory;
+import tachyon.underfs.UnderFileSystemRegistry;
+
+/**
+ * This test ensures the s3 ufs module correctly accepts paths that begin with s3n://
```
Comment (calvinjia) [inline underfs/local/src/test/java/tachyon/underfs/local/LocalUnderFileSystemFactoryTest.java]: This description does not seem accurate?

File: underfs/local/src/test/resources/log4j.properties [hunk 3]
```diff
@@ -0,0 +1,17 @@
+tachyon.root.logger=INFO, TEST_LOGGER
+tachyon.log.dir=.
+tachyon.log.file=logs/tests.log
```
Comment (calvinjia) [inline underfs/local/src/test/resources/log4j.properties]: Could we change the logging dir to be consistent with the other log4j test properties? I think the idea is to have mvn clean cleanup the test logs as well.

File: underfs/local/pom.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    <dependency>
+      <groupId>org.powermock</groupId>
+      <artifactId>powermock-api-mockito</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.powermock</groupId>
+      <artifactId>powermock-reflect</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (calvinjia) [inline underfs/local/pom.xml]: Do we need the `powermock` dependencies?

File: examples/pom.xml [hunk 5]
```diff
@@ -45,6 +45,14 @@
...
-    </dependency>    
+    </dependency>
+
+    <!-- Test Dependencies -->
+    <dependency>
+      <groupId>org.tachyonproject</groupId>
+      <artifactId>tachyon-underfs-local</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (calvinjia) [inline examples/pom.xml]: I don't think there are any tests in the `examples` module?

File: servers/pom.xml [hunk 6]
```diff
@@ -117,6 +117,12 @@
       <artifactId>powermock-api-mockito</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.tachyonproject</groupId>
+      <artifactId>tachyon-underfs-local</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (calvinjia) [inline servers/pom.xml]: Do the unit tests here require the local ufs?

**Comment by calvinjia** [comment]
@apc999 Thanks for the patch, I did a pass over the changes.

File: examples/pom.xml (hunk 5 above)
Comment (apc999) [inline examples/pom.xml]: Good catch. Removed.

File: integration-tests/pom.xml (hunk 1 above)
Comment (apc999) [inline integration-tests/pom.xml]: This is for `TestUtils` in `tachyon-servers`. We should remove this dependency. A TODO added as comment.

File: servers/pom.xml (hunk 6 above)
Comment (apc999) [inline servers/pom.xml]: AllocateStrategyTest, EvictStrategyTest, StorageDirTest, StorageTierTest need ufs factory to recognize localfs.
Maybe move them down to integration tests?

File: underfs/local/pom.xml (hunk 4 above)
Comment (apc999) [inline underfs/local/pom.xml]: this was somehow included indirectly before, I added it due to mvn dependency:analyze

File: underfs/local/src/test/java/tachyon/underfs/local/LocalUnderFileSystemFactoryTest.java (hunk 2 above)
Comment (apc999) [inline underfs/local/src/test/java/tachyon/underfs/local/LocalUnderFileSystemFactoryTest.java]: Fixed. Thanks

File: underfs/local/src/test/resources/log4j.properties (hunk 3 above)
Comment (apc999) [inline underfs/local/src/test/resources/log4j.properties]: Fixed. Thanks

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**